cvmfs/kvstore.h

```diff
@@ -0,0 +1,208 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_KVSTORE_H_
+#define CVMFS_KVSTORE_H_
+
+#include <pthread.h>
+#include <unistd.h>
+
+#include <string>
+
+#include "cache.h"
+#include "lru.h"
+#include "statistics.h"
+
+using namespace std;  // NOLINT
+
+namespace kvstore {
+
+struct MemoryBuffer {
+  void *address;
+  size_t size;
+  unsigned int refcount;
+  cache::CacheManager::ObjectType object_type;
```

Comment:
You're right, I think we want to keep it... if only to allow for proper reporting of things in the cache.